### ISSUE TITLE
In E5, what was the size of m(mini-batch size)?

### ISSUE THREAD

**Issue by quovadisss** [microsoft/unilm#1448]
I read 'Text Embeddings by Weakly-Supervised Contrastive Pre-training' (https://arxiv.org/pdf/2212.03533.pdf) and would like to know the size of the mini-batch (m). Additionally, I'm interested in the size of the dataset used for pre-training. If the mini-batch size is 32, does it mean that 32 query-positive pairs are assigned to one device? I couldn’t find information about the mini-batch size in the paper.

Thank you very much.

**Comment by intfloat** [comment]
Hi @quovadisss ,

The batch size information is available in Table 11 of Appendix. For pre-training, the batch size is 32k (aggregated across all GPUs). The per device batch size is `32k / # of GPUs` (16 / 32 / 64 for small / base / large models).

**Comment by quovadisss** [comment]
Thank you for your answer @intfloat.

I understand that the total batch size is 32k. But I want to know the size of m(length of a list of negative passages) for the i-th example. In the paper, I found that you used in-batch negative sampling for contrastive pre-training. I want to know how many negative passages that you used for the i-th example. Did you sample negatives randomly in the in-batch?

**Comment by intfloat** [comment]
The in-batch negatives are all other documents in the same batch (m is `32k - 1`).

**Comment by quovadisss** [comment]
Do you mean that the i-th example consists of a query, a positive, and 31,999 negatives? You mentioned that the 32,000 examples are distributed across 16, 32, or 64 devices, which means each device would handle a maximum of 2,000 examples. I also noticed you're using gradient accumulation steps due to memory constraints, suggesting that the number of negatives should be fewer than 2,000. Additionally, I think 2,000 negatives might be excessive. I doubt that learning with so many negatives would be effective. I'm not sure if my reasoning is correct.
